lib/miq_automation_engine/models/miq_git_repo.rb

```diff
@@ -0,0 +1,60 @@
+class MiqGitRepo < ApplicationRecord
+  include AuthenticationMixin
+  serialize :branches
+  serialize :tags
+  validates_presence_of :url
+  validates :url, :format => URI::regexp(%w(http https))
+  before_save :default_values
+
+  def verify_credentials(_auth_type = nil, _options = {})
+    refresh
+  end
+
+  def refresh
+    repo = Dir.exist?(dirname) ? update_repo : init_repo
+    self.branches = repo.branches
+    self.tags = repo.tags
+    save!
+  end
+
+  private
+
+  def init_repo
+    with_connection do
+      params = {:clone => true, :url => url, :path => dirname}
+      params[:ssl_no_verify] = verify_ssl.nil? || verify_ssl == 0
+      if authentications.any?
+        params[:username] = authentications.first.userid
+        params[:password] = authentications.first.password
+      end
+      GitWorktree.new(params)
+    end
+  end
+
+  def update_repo
+    with_connection do
+      GitWorktree.new(:path => dirname).tap do |repo|
+        repo.send(:fetch_and_merge)
+      end
+    end
+  end
+
+  def with_connection
```

Comment:
I believe ActiveRecord already has a with_connection method, so we might want to avoid calling it that.